vitest.config.ts

```diff
@@ -0,0 +1,29 @@
+import { defineConfig } from "vitest/config";
+
+export default defineConfig({
+  test: {
+    exclude: ["**/*.{integ,e2e,browser}.spec.{ts,js}"],
+    include: [
+      "lib/lib-dynamodb/**/*.spec.{ts,js}",
+      "lib/lib-storage/**/*.spec.{ts,js}",
+      // "packages/body-checksum-browser/**/*.spec.{ts,js}",
+      "packages/body-checksum-node/**/*.spec.{ts,js}",
+      "packages/chunked-stream-reader-node/**/*.spec.{ts,js}",
+      "packages/cloudfront-signer/**/*.spec.{ts,js}",
+      "packages/core/**/*.spec.{ts,js}",
+      "packages/credential-provider-cognito-identity/**/*.spec.{ts,js}",
+      "packages/credential-provider-env/**/*.spec.{ts,js}",
+      "packages/credential-provider-http/**/*.spec.{ts,js}",
+      "packages/credential-provider-ini/**/*.spec.{ts,js}",
+      "packages/credential-provider-node/**/*.spec.{ts,js}",
+      "packages/credential-provider-process/**/*.spec.{ts,js}",
+      "packages/credential-provider-sso/**/*.spec.{ts,js}",
+      "packages/credential-provider-web-identity/**/*.spec.{ts,js}",
+      "packages/credential-providers/**/*.spec.{ts,js}",
+      "packages/ec2-metadata-service/**/*.spec.{ts,js}",
+      "packages/endpoint-cache/**/*.spec.{ts,js}",
+      "packages/eventstream-handler-node/**/*.spec.{ts,js}",
+    ],
+    environment: "node",
+  },
+});
```

Comment:
This config file explicitly lists out packages because not all .spec.ts files are vitest compatible.

In CI, `test:unit` is run by turbo, so this file is not used, and does not need to be maintained. When conversion is complete, the inclusion will be wild-card based instead of curated, and only used if a developer wishes to run `npx vite run` etc. from the root.